impeller/entity/contents/filters/gaussian_blur_filter_contents.cc

```diff
@@ -0,0 +1,273 @@
+                                        SamplerAddressMode address_mode) {
+  SamplerDescriptor sampler_desc;
+  sampler_desc.min_filter = filter;
+  sampler_desc.mag_filter = filter;
+  sampler_desc.width_address_mode = address_mode;
+  sampler_desc.width_address_mode = address_mode;
+  return sampler_desc;
+}
+
+template <typename T>
+void BindVertices(Command& cmd,
+                  HostBuffer& host_buffer,
+                  std::initializer_list<typename T::PerVertexData>&& vertices) {
+  VertexBufferBuilder<typename T::PerVertexData> vtx_builder;
+  vtx_builder.AddVertices(vertices);
+  auto vtx_buffer = vtx_builder.CreateVertexBuffer(host_buffer);
+  cmd.BindVertices(vtx_buffer);
+}
+
+std::shared_ptr<Texture> MakeDownsampleSubpass(
+    const ContentContext& renderer,
+    std::shared_ptr<Texture> input_texture,
+    const SamplerDescriptor& sampler_descriptor,
+    const std::array<Point, 4>& uvs,
+    const ISize& subpass_size) {
+  ContentContext::SubpassCallback subpass_callback =
+      [&](const ContentContext& renderer, RenderPass& pass) {
+        HostBuffer& host_buffer = pass.GetTransientsBuffer();
+
+        Command cmd;
+        DEBUG_COMMAND_INFO(cmd, "Gaussian blur downsample");
+        auto pipeline_options = OptionsFromPass(pass);
+        pipeline_options.primitive_type = PrimitiveType::kTriangleStrip;
+        cmd.pipeline = renderer.GetTexturePipeline(pipeline_options);
+
+        TextureFillVertexShader::FrameInfo frame_info;
+        frame_info.mvp = Matrix::MakeOrthographic(ISize(1, 1));
+        frame_info.texture_sampler_y_coord_scale = 1.0;
+        frame_info.alpha = 1.0;
+
+        BindVertices<TextureFillVertexShader>(cmd, host_buffer,
+                                              {
+                                                  {Point(0, 0), uvs[0]},
+                                                  {Point(1, 0), uvs[1]},
+                                                  {Point(0, 1), uvs[2]},
+                                                  {Point(1, 1), uvs[3]},
+                                              });
+
+        TextureFillVertexShader::BindFrameInfo(
+            cmd, host_buffer.EmplaceUniform(frame_info));
+        TextureFillFragmentShader::BindTextureSampler(
+            cmd, input_texture,
+            renderer.GetContext()->GetSamplerLibrary()->GetSampler(
+                sampler_descriptor));
+
+        pass.AddCommand(std::move(cmd));
+
+        return true;
+      };
+  std::shared_ptr<Texture> out_texture = renderer.MakeSubpass(
+      "Gaussian Blur Filter", subpass_size, subpass_callback);
+  return out_texture;
+}
+
+std::shared_ptr<Texture> MakeBlurSubpass(
+    const ContentContext& renderer,
+    std::shared_ptr<Texture> input_texture,
+    const SamplerDescriptor& sampler_descriptor,
+    const GaussianBlurFragmentShader::BlurInfo& blur_info) {
+  // TODO(gaaclarke): This doesn't render if the blur_info.sigma == 0.
```

Comment:
Where do we check for sigma < kEhClose enough on the old blur? Presumibly we could detect that and just make the filter return the original contents, or "optimize" it out entirely

Reply:
Good call.  I do have a todo to address this.  It didn't occur to me yet that the solution was so simple, done.